clients/drcachesim/tracer/raw2trace.cpp

```diff
@@ -894,55 +944,36 @@ raw2trace_t::process_syscall_pt(raw2trace_thread_data_t *tdata, uint64_t syscall
         tdata->pt_metadata_processed = true;
     }
 
-    if (tdata->pre_read_pt_entries.empty() && tdata->kthread_file->eof()) {
+    if (pt_data == nullptr) {
+        if (!tdata->error.empty())
+            return false;
         VPRINT(1, "Finished decoding all PT data for thread %d\n", tdata->tid);
         return true;
     }
+    uint64_t syscall_args_num =
+        pt_data->header[PDB_HEADER_NUM_ARGS_IDX].syscall_args_num.args_num;
+    uint64_t pt_data_size =
+        pt_data->header[PDB_HEADER_DATA_BOUNDARY_IDX].pt_data_boundary.data_size -
+        PT_DATA_PDB_HEADER_SIZE - syscall_args_num * sizeof(uint64_t);
 
-    if (tdata->pre_read_pt_entries.empty()) {
-        syscall_pt_entry_t header[PT_DATA_PDB_HEADER_ENTRY_NUM];
-        if (!tdata->kthread_file->read((char *)&header[0], PT_DATA_PDB_HEADER_SIZE)) {
-            if (tdata->kthread_file->eof()) {
-                VPRINT(1, "Finished decoding all PT data for thread %d\n", tdata->tid);
-                return true;
-            }
-            tdata->error = "Unable to read the PDB header of next syscall's PT data "
-                           "form kernel thread log file";
-            return false;
-        }
-        tdata->pre_read_pt_entries.insert(tdata->pre_read_pt_entries.end(), header,
-                                          header + PT_DATA_PDB_HEADER_ENTRY_NUM);
-    }
-
-    if (tdata->pre_read_pt_entries[PDB_HEADER_DATA_BOUNDARY_IDX].pt_data_boundary.type !=
-            SYSCALL_PT_ENTRY_TYPE_PT_DATA_BOUNDARY ||
-        tdata->pre_read_pt_entries[PDB_HEADER_SYSCALL_IDX_IDX].syscall_idx.type !=
-            SYSCALL_PT_ENTRY_TYPE_SYSCALL_IDX ||
-        tdata->pre_read_pt_entries[PDB_HEADER_SYSCALL_IDX_IDX].syscall_idx.idx !=
-            syscall_idx) {
```

Comment:
Where did these checks go?

Reply:
Boundary check continues to be here. idx check is just below.